lsp/README

```diff
@@ -240,9 +240,20 @@ checks and does not show any preview of the changes so it is best to use this
 feature only after committing all modified files so you can
 easily revert to a working state if needed. Since this is potentially a
 dangerous operation, to prevent accidental renames, the "Rename" button in the
-dialog is not selected by defalut and simply pressing enter just cancels the
+dialog is not selected by default and simply pressing enter just cancels the
 dialog.
 
+Performance implications
+========================
+
+To provide all the features, this plugin needs to react on various events
+like keystrokes, mouse movement, clicks, etc. and also communicate heavily
+with the LSP server to send the current document text and what kind of event
+happened.
+These additional tasks cause more CPU usage of the Geany process and especially
+the LSP server process, so they require also more power which might be
```

Comment:
```
and especially LSP server process
```

From what I have seen, this is the main source of higher battery drain. Especially with `clangd` which recompiles the file on every key stroke (you can see this in stderr when you enable it in the config file). Also `pylsp` is ultra-slow.

Some servers may also spend too much time on certain tasks - for instance, semantic tokens and document symbols may be expensive. It might help to disable these in the config file if it's an issue. No matter what, though, the server will always have to re-parse the current source on modifications.

Reply:
Yeah, I gave up on `pylsp`, after I got it working also with proper virtualenv integration and everything, it was sooooo slow that I had to wait 5-10s for a calltip, autocompletion was often even slower and the `pylsp` process burned one CPU core all the time.

I switched to your suggestion: lsp-proxy + jedi-lsp + ruff-lsp.
This will probably work well enough.

Thanks for your suggestions!

Reply:
> it was sooooo slow that I had to wait 5-10s for a calltip

This is much worse than I thought - with smaller files it seems to work reasonably well. But it's definitely the slowest server I tried.